src/libs/Trie/index.js

```diff
@@ -0,0 +1,121 @@
+import _ from 'underscore';
+import TrieNode from './TrieNode';
+
+class Trie {
+    constructor() {
+        this.root = new TrieNode();
+    }
+
+    /**
+    * Add a word to the Trie
+    * @param {String} word
+    * @param {Object} [metaData] - attach additional data to the word
+    * @param {TrieNode} newNode
+    * @param {Boolean} [containChar] - empty word doesn't have any char
+    * @returns {void}
+    */
+    add(word, metaData, newNode = this.root, containChar = false) {
+        const node = newNode;
+        if (word.length === 0 && !containChar) {
+            throw new Error('Cannot insert empty word into Trie :', word);
```

Comment:
No point in including a word we know is empty:

```suggestion
            throw new Error('Cannot insert empty word into Trie');
```

Reply:
Done.